src/grading-settings/assignment-section/assignments/AssignmentTypeName.jsx

```diff
@@ -0,0 +1,70 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { FormattedMessage, injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { Form } from '@edx/paragon';
+
+import { ASSIGNMENT_TYPES } from '../utils/enum';
+import messages from '../messages';
+
+const AssignmentTypeName = ({
+  intl, value, errorEffort, onChange,
+}) => {
+  const initialAssignmentName = useRef(value);
+
+  return (
+    <li className="course-grading-assignment-type-name">
+      <Form.Group className={classNames('form-group-custom', {
+        'form-group-custom_isInvalid': errorEffort,
+      })}
+      >
+        <Form.Label className="grading-label">
+          {intl.formatMessage(messages.assignmentTypeNameTitle)}
+        </Form.Label>
+        <Form.Control
+          data-testid="assignment-type-name-input"
+          type="text"
+          name={ASSIGNMENT_TYPES.type}
+          onChange={onChange}
+          value={value}
+          isInvalid={Boolean(errorEffort)}
+        />
+        <Form.Control.Feedback className="grading-description">
+          {intl.formatMessage(messages.assignmentTypeNameDescription)}
+        </Form.Control.Feedback>
+        {errorEffort && errorEffort !== 'duplicateAssignmentName' && (
```

Comment:
may we have a constant for `duplicateAssignmentName`?
I see we use it in different places...

Reply:
Corrected, thanks